arwen/contexts/runtime.go

```diff
@@ -833,6 +838,31 @@ func (context *runtimeContext) GetAllErrors() error {
 	return context.errors
 }
 
+// EnableTraceGas sets true to the flag variable that traces gas consumption
+func (context *runtimeContext) EnableTraceGas() {
+	context.traceGasEnabled = true
+}
+
+// DisableTraceGas sets true to the flag variable that traces gas consumption
+func (context *runtimeContext) DisableTraceGas() {
+	context.traceGasEnabled = false
+}
+
+// TraceGasUsed adds the usedGas passed to the traced gas map with the key as the passed functionName
+func (context *runtimeContext) TraceGasUsed(functionName string, usedGas uint64) {
```

Comment:
The runtime context has access to the current gas left. Could we also create a method that only takes the initial gas and computes the delta itself? So we don't have to write so much in each EI function?